src/extent.c

```diff
@@ -1105,6 +1126,13 @@ extent_grow_retained(tsdn_t *tsdn, arena_t *arena,
 		goto label_err;
 	}
 
+	/*
+	 * There are situations in which the Linux kernel will include
```

Comment:
Be more specific here, especially with steps to repro (I hate when I find comments like this years later and don't know if I can remove or modify it...)

Linux *always* includes the mapped memory in dumps (except with MADV_DONTDUMP), it just happens to be sparsified with modern filesystems, and not with piped core_patterns